compiler/modules/front/expr.bal

```diff
@@ -1018,6 +1020,21 @@ function validArgumentCount(ExprContext cx, bir:FunctionRef func, int nSuppliedA
     }
 }
 
+function validArgumentTypes(ExprContext cx, bir:FunctionRef func, bir:Operand[] suppliedArgs, s:MethodCallExpr|s:FunctionCallExpr expr) returns CodeGenError? {
+    foreach int i in 0 ..< suppliedArgs.length() {
+        if operandHasType(cx.mod.tc, suppliedArgs[i], func.signature.paramTypes[i]) {
+            continue;
+        }
+        return cx.semanticErr(`wrong argument type for parameter ${i + 1} in call to function ${symbolToString(cx.mod, cx.defn.part.partIndex, func.symbol)}`, expr.args[i].startPos);
```

Comment:
This won't work because method arg expression index is one less than expression arg expression index.

Reply:
Also use a range for the argument expression.